### PR TITLE
feat: add darwin quickjs support

### PR DESCRIPTION
**Pull request by gengjiawen**
Fix https://github.com/devsnek/esvu/issues/18

### CODE REVIEW

File: src/engines/quickjs.js [hunk 1]
```diff
@@ -47,9 +50,13 @@ class QuickJSInstaller extends Installer {
...
+      if (platform.startsWith('darwin')) {
+        this.binPath = await this.registerBinary('quickjs');
+      } else {
+        this.binPath = await this.registerBinary('qjs', 'quickjs');
+        // for eshost
+        await this.registerBinary('run-test262', 'quickjs-run-test262');
+      }
     } else {
       await this.registerAsset('libwinpthread-1.dll');
       const qjs = await this.registerAsset('qjs.exe');
```
Comment (devsnek) [inline src/engines/quickjs.js:62]:
```suggestion
if (platform === 'darwin-x64') {
  this.binPath = await this.registerBinary('quickjs');
} else if (platform.startsWith('win')) {
  await this.registerAsset('libwinpthread-1.dll');
  const qjs = await this.registerAsset('qjs.exe');
  this.binPath = await this.registerScript('qjs', `"${qjs}"`);
} else {
  this.binPath = await this.registerBinary('qjs', 'quickjs');
  await this.registerBinary('run-test262', 'quickjs-run-test262');
}
```

File: src/engines/quickjs.js [hunk 2]
```diff
@@ -38,6 +38,9 @@ class QuickJSInstaller extends Installer {
   }
 
   getDownloadURL(version) {
+    if (platform.startsWith('darwin')) {
```
Comment (devsnek) [inline src/engines/quickjs.js:41]:
```suggestion
    if (platform === 'darwin-x64') {
```

**Comment by devsnek** [comment]
Do your builds not include `run-test262`? Without that binary this is kind of useless, since eshost needs it.

**Comment by gengjiawen** [comment]
> Do your builds not include `run-test262`? Without that binary this is kind of useless, since eshost needs it.

Let's wait https://github.com/napi-bindings/quickjs-build/pull/8 then. cc @NickNaso

**Comment by gengjiawen** [comment]
@devsnek PTAL.

**Pull request merged**